pkg/resolvers/k8s/watcher.go

```diff
@@ -54,119 +51,136 @@ func (w *watcher) Close() {
 func (w *watcher) Next() ([]*naming.Update, error) {
 	if w.ctx.Err() != nil {
 		// We already stopped.
-		return []*naming.Update(nil), errors.Wrap(w.ctx.Err(), "k8sresolver: watcher.Next already stopped or Next returned error already. "+
+		return []*naming.Update(nil), errors.Wrap(w.ctx.Err(), "watcher.Next already stopped or Next returned error already. "+
 			"Note that watcher errors are not recoverable.")
 	}
 	u, err := w.next()
 	if err != nil {
 		// Just in case.
 		w.Close()
+		return u, errors.Wrap(err, "k8sresolver")
```

Comment:
is this rather terse error message intentional?

Reply:
yes, it adds context where this errors came from.